### PR TITLE
Document the linear() easing function

### PR DESCRIPTION
**Pull request by dipikabh**
#### Summary

- `linear()` function is supported in FF104 behind a preference: `layout.css.linear-easing-function.enabled`.
- This PR updates the [`<easing-function>`](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function) page to add the `linear()` function.
- In addition to adding the `linear()` function to 'Syntax' and 'Description' sections, this PR makes the following changes:
    - Moves the beginning prose to the section on 'Cubic Bézier functions' because the same information was being repeated in two places.
    - Creates an H3 'Values' section after 'Syntax' (to be consistent with the order and style in other CSS reference pages) and moves the value descriptions from the function sections to the 'Values' section. 
- There is scope for explaining the `linear()` syntax better (input and output values, percentages) and adding valid and invalid examples to 'Examples' section like the ones for cubic-bezier() and steps(). I'll tackle this in the next round.
> **UPDATE:** Opened https://github.com/mdn/content/issues/20138 to track the improvement work

#### Supporting details

Bug tracker: https://bugzilla.mozilla.org/show_bug.cgi?id=1764126
Spec: https://drafts.csswg.org/css-easing-2/#the-linear-easing-function

#### Related issues and PRs

Doc tracker: https://github.com/mdn/content/issues/19709
PR for BCD update: https://github.com/mdn/browser-compat-data/pull/17527
PR for experimental feature update: https://github.com/mdn/content/pull/19671

#### Metadata

- [ ] Adds a new document
- [X] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/css/easing-function/index.md [hunk 1]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
 
-#### Keywords for common cubic-bezier easing functions
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Two dots are present, the first at the X 0 Y 0 position, and the second at the X 1 Y 1 position. The first dot has a horizontal lines extending 1 unit forwards away from the Y axis.](steps-1-end.png)
 
-- `ease`
-  - : The interpolation starts slowly, accelerates sharply, and then slows gradually towards the end. This keyword represents the easing function `cubic-bezier(0.25, 0.1, 0.25, 1.0)`. It is similar to [`ease-in-out`](#ease-in-out), though it accelerates more sharply at the beginning.
+## Description
 
-![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A curving line extends from the origin to the X 1 Y 1 position, quickly rising and arcing.](cubic-bezier-ease.png)
+The are three types of easing functions:
```
Comment (Rumyra) [inline files/en-us/web/css/easing-function/index.md:97]:
```suggestion
There are three types of easing functions:
```

**Review by Rumyra — DISMISSED** [review]

File: files/en-us/web/css/easing-function/index.md [hunk 2]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
+  - : {{cssxref("&lt;number&gt;")}} values representing the abscissas and ordinates of the [P1 and P2 points](#cubic_bézier_functions) defining the cubic Bézier curve. x1 and x2 must be in the range `[0, 1]`, otherwise the value is invalid.
 
-The interpolation is done at a constant rate from beginning to end. Use the `linear` keyword for this function. This represents the easing function `cubic-bezier(0.0, 0.0, 1.0, 1.0)`.
+- `ease`
+  - : Indicates that the interpolation starts slowly, accelerates sharply, and then slows gradually towards the end. This keyword represents the easing function `cubic-bezier(0.25, 0.1, 0.25, 1.0)`. It is similar to [`ease-in-out`](#ease-in-out), though it accelerates more sharply at the beginning.
 
-![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A straight diagonal line extends from the origin to the X 1 Y 1 position.](cubic-bezier-linear.png)
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A curving line extends from the origin to the X 1 Y 1 position, quickly rising and arcing.](cubic-bezier-ease.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:56]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio' with a curved line quickly rising from the origin to the X 1 Y 1 position.](cubic-bezier-ease.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 3]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
 
-### Cubic Bézier functions
+- `ease-in`
+  - : Indicates that the interpolation starts slowly, then progressively speeds up until the end, at which point it stops abruptly. This keyword represents the easing function `cubic-bezier(0.42, 0.0, 1.0, 1.0)`.
 
-The `cubic-bezier()` functional notation defines a [cubic Bézier curve](/en-US/docs/Glossary/Bezier_curve). As these curves are continuous, they are often used to smooth down the start and end of the interpolation and are therefore sometimes called _easing functions_.
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A slightly curving line extends from the origin to the X 1 Y 1 position, with most of the curve close to the origin, straightening out as it approaches X 1 Y 1.](cubic-bezier-ease-in.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:61]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio' shows a shallow curve from the origin that straightens out as it approaches X1 Y1.](cubic-bezier-ease-in.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 4]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
+- `ease-in-out`
+  - : Indicates that the interpolation starts slowly, speeds up, and then slows down towards the end. This keyword represents the easing function `cubic-bezier(0.42, 0.0, 0.58, 1.0)`. At the beginning, it behaves like the [`ease-in`](#ease-in) function; at the end, it is like the [`ease-out`](#ease-out) function.
 
-A cubic Bézier curve is defined by four points P0, P1, P2, and P3. P0 and P3 are the start and the end of the curve and, in CSS these points are fixed as the coordinates are ratios (the abscissa the ratio of time, the ordinate the ratio of the output range). P0 is `(0, 0)` and represents the initial time or position and the initial state, P3 is `(1, 1)` and represents the final time or position and the final state.
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A slightly curving line extends from the origin to the X 1 Y 1 position. The curve is symmetrical, resembling a stretched out letter S.](cubic-bezier-ease-in-out.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:66]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio' shows a symmetrical, 'S'-shaped curving line from the origin to the X1 Y1 position.](cubic-bezier-ease-in-out.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 5]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
 
-Not all cubic Bézier curves are suitable as easing functions as not all are [mathematical functions](https://en.wikipedia.org/wiki/Function_%28mathematics%29); i.e., curves that for a given abscissa have zero or one value. With P0 and P3 fixed as defined by CSS, a cubic Bézier curve is a function, and is therefore valid, if and only if the abscissas of P1 and P2 are both in the `[0, 1]` range.
+- `ease-out`
+  - : Indicates that the interpolation starts abruptly and then progressively slows down towards the end. This keyword represents the easing function `cubic-bezier(0.0, 0.0, 0.58, 1.0)`.
 
-Cubic Bézier curves with the P1 or P2 ordinate outside the `[0, 1]` range may generate _bouncing_ effects.
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A slightly curving line extends from the origin to the X 1 Y 1 position, starting as an almost straight diagonal line and curving as it gets close to X 1 Y 1.](cubic-bezer-ease-out.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:71]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio' shows a straight diagonal line that lightly curves as it gets close to X1 Y1.](cubic-bezer-ease-out.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 6]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
+    - `end` is the equivalent of `jump-end`. This is the default.
 
-```css
-cubic-bezier(x1, y1, x2, y2)
-```
+- `step-start`
+  - : Indicates that the interpolation jumps immediately to its final state, where it stays until the end. This keyword represents the easing function `steps(1, jump-start)` or `steps(1, start)`.
 
-where:
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Two dots are present, the first at the X 0 Y 0 position, and the second at the X 1 Y 1 position. The second dot has a horizontal lines extending 1 units back towards the Y axis.](steps-1-start.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:88]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio'. There is a point at X0 Y0 and another at X1 Y1 with a line extending towards the Y axis.](steps-1-start.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 7]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
 
-- **x1, y1, x2, y2**
-  - : Are {{cssxref("&lt;number&gt;")}} values representing the abscissas, and ordinates of the P1 and P2 points defining the cubic Bézier curve. x1 and x2 must be in the range \[0, 1] or the value is invalid.
+- `step-end`
+  - : Indicates that the interpolation stays in its initial state until the end, at which point it jumps directly to its final state. This keyword represents the easing function `steps(1, jump-end)` or `steps(1, end)`.
 
-#### Keywords for common cubic-bezier easing functions
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Two dots are present, the first at the X 0 Y 0 position, and the second at the X 1 Y 1 position. The first dot has a horizontal lines extending 1 unit forwards away from the Y axis.](steps-1-end.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:93]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio'. There is a point at X0 Y0 with a line extending away from the Y axis and a point at X1 Y1.](steps-1-end.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 8]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
+These are a few examples illustrating the `steps()` function:
 
-- `steps(3, jump-both)`
+```css
+steps(2, jump-start)
+steps(2, start)
+```
 
-  ![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Five dots are present, the first at the X 0 Y 0 position, the second at the X 0 Y 0.25 position, the third at the X 0.5 Y 0.5 position, the fourth at the X 0.75 Y 0.75 position, and the fifth at the X 1 Y 1 position. The second, third, and fourth dots have horizontal lines extending 0.25 units forwards away from the Y axis.](step3both.png)
+  ![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Three dots are present, the first at the X 0 Y 0 position, the second at the X 0.5 Y 0.5 position, and the third at the X 1 Y 1 position. The second and third dots have horizontal lines extending 0.5 units backwards towards the Y axis.](steps-2-start.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:152]:
```suggestion
  ![A 2D graph of 'Time ratio' to 'Output ratio'. There are points at X0 Y0, X0.5 Y0.5, and X1 Y1. The second and third points have lines towards the Y axis.](steps-2-start.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 9]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
-- `step-end`
-  - : The interpolation stays in its initial state until the end, at which point it jumps directly to its final state. This keyword represents the easing function `steps(1, jump-end)` or `steps(1, end)`.
+  ![Five steps, with no jumps, so 20% of the time is at the beginning state or 0% mark and the last 20% is at the final state, or 100% mark.](step5none.png)
 
-![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Two dots are present, the first at the X 0 Y 0 position, and the second at the X 1 Y 1 position. The first dot has a horizontal lines extending 1 unit forwards away from the Y axis.](steps-1-end.png)
+```css
+steps(3, jump-both)
+```
+
+  ![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. Five dots are present, the first at the X 0 Y 0 position, the second at the X 0 Y 0.25 position, the third at the X 0.5 Y 0.5 position, the fourth at the X 0.75 Y 0.75 position, and the fifth at the X 1 Y 1 position. The second, third, and fourth dots have horizontal lines extending 0.25 units forwards away from the Y axis.](step3both.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:171]:
```suggestion
  ![A 2D graph of 'Time ratio' to 'Output ratio'. There are points in 0.25 steps on both axes from X0 Y0 to X1 Y1. The second, third, and fourth points have lines extending 0.25 units away from the Y axis.](step3both.png)
```

File: files/en-us/web/css/easing-function/index.md [hunk 10]
```diff
@@ -17,163 +17,170 @@ browser-compat: css.types.easing-function
...
 
-- `ease-out`
-  - : The interpolation starts abruptly, and then progressively slows down towards the end. This keyword represents the easing function `cubic-bezier(0.0, 0.0, 0.58, 1.0)`.
+Consider another example of the function: `linear(0, 0.25 75%, 1)`. This produces a linear easing function that spends 75% of the time transitioning from `0` to `.25` and the last 25% transitioning from `.25` to `1`.
 
-![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A slightly curving line extends from the origin to the X 1 Y 1 position, starting as an almost straight diagonal line and curving as it gets close to X 1 Y 1.](cubic-bezer-ease-out.png)
+The `linear` keyword produces a `linear()` function with two points. This is equivalent to the easing function `cubic-bezier(0.0, 0.0, 1.0, 1.0)`.
 
-### Step functions
+![A graph with X and Y ranges from 0 to 1, with the X axis labeled 'Time ratio' and the Y axis labeled 'Output ratio'. A straight diagonal line extends from the origin to the X 1 Y 1 position.](cubic-bezier-linear.png)
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/index.md:113]:
```suggestion
![A 2D graph of 'Time ratio' to 'Output ratio'. A straight diagonal line extends from the origin to the X1 Y1 position.](cubic-bezier-linear.png)
```

**Comment by bsmth** [comment]
minor: the bezier gif doesn't seem to load on the preview, looks okay on prod though 🤔 :

* https://pr19896.content.dev.mdn.mozit.cloud/en-US/docs/Glossary/Bezier_curve/b%C3%A9zier_2_big.gif
* https://developer.mozilla.org/en-US/docs/Glossary/Bezier_curve/b%C3%A9zier_2_big.gif

**Comment by dipikabh** [comment]
Thanks @bsmth for taking the time to review (and shorten) all the alt texts! Hopefully this covers all of them on this page.

I was wondering though if the single quotes as in 'Time ratio' are needed in an alt text.

To your comment:
> minor: the bezier gif doesn't seem to load on the preview, looks okay on prod though 🤔 :
> 
>     https://pr19896.content.dev.mdn.mozit.cloud/en-US/docs/Glossary/Bezier_curve/b%C3%A9zier_2_big.gif
>     https://developer.mozilla.org/en-US/docs/Glossary/Bezier_curve/b%C3%A9zier_2_big.gif

🤷‍♀️ maybe the local build does not support gifs?

**Comment by bsmth** [comment]
> I was wondering though if the single quotes as in 'Time ratio' are needed in an alt text.

Feel free to remove / reword 👍🏻 

>maybe the local build does not support gifs?

Yeah it looks like this will work when deployed, the asset URL looks correct

**Review by bsmth — APPROVED** [review]
Looks good 👍🏻

**Comment by bsmth** [comment]
Merging, thank you 🙌🏻

**Pull request merged**